File: com.facadepattern/python/MultimediaFacade.py
from MultimediaPlayer import AudioPlayer, VideoPlayer, ImageLoader

class MultimediaFacade:
    audioPlayer: AudioPlayer
    videoPlayer: VideoPlayer
    imageLoader: ImageLoader
    
    def __init__(self) -> None:
        self.audioPlayer = AudioPlayer()
        self.videoPlayer = VideoPlayer()
        self.imageLoader = ImageLoader()
        
    def playMedia(self, filename: str, media_type: str) -> None:
        if (media_type == 'audio'):
            self.audioPlayer.playAudio(filename)
        elif (media_type == 'video'):
            self.videoPlayer.playVideo(filename)
        elif (media_type == 'image'):
            self.imageLoader.loadImage(filename)
        else:
            print(f'Invalid media type: {media_type}')

File: Facade/java/MultimediaFacade.java
import com.facadepattern.java.AudioPlayer;
import com.facadepattern.java.VideoPlayer;
import com.facadepattern.java.ImageLoader;

// Facade class
class MultimediaFacade {
    private AudioPlayer audioPlayer;
    private VideoPlayer videoPlayer;
    private ImageLoader imageLoader;

    public MultimediaFacade() {
        this.audioPlayer = new AudioPlayer();
        this.videoPlayer = new VideoPlayer();
        this.imageLoader = new ImageLoader();
    }

    void playMedia(String filename, String mediaType) {
        if (mediaType.equals("audio")) {
            audioPlayer.playAudio(filename);
        } else if (mediaType.equals("video")) {
            videoPlayer.playVideo(filename);
        } else if (mediaType.equals("image")) {
            imageLoader.loadImage(filename);
        } else {
            System.out.println("Unsupported media type " + mediaType);
        }
    }
}